src/ray/object_manager/object_manager.cc

```diff
@@ -201,8 +197,18 @@ void ObjectManager::TryPull(const ObjectID &object_id) {
 
   RAY_LOG(DEBUG) << "Sending pull request from " << client_id_ << " to " << client_id
                  << " of object " << object_id;
-  // Try pulling from the client.
-  PullEstablishConnection(object_id, client_id);
+
+  auto rpc_client = GetRpcClient(client_id);
+  if (rpc_client) {
+    // Try pulling from the client.
+    rpc_service_.post([this, object_id, client_id, rpc_client]() {
+      auto st = SendPullRequest(object_id, client_id, rpc_client);
```

Comment:
this status should be checked

Reply:
i think it's always ok, the exception would be handled by client callback(at present, only a warning is shown)

Reply:
then make the function return void?